pandas/core/indexes/period.py

```diff
@@ -230,7 +230,10 @@ def _simple_new(cls, values: PeriodArray, name: Label = None):
             Values that can be converted to a PeriodArray without inference
             or coercion.
         """
-        assert isinstance(values, PeriodArray), type(values)
+        if not isinstance(values, PeriodArray):
+            raise TypeError(
+                f"_simple_new expects PeriodArray, got {type(values).__name__}"
+            )
```

Comment:
I think an assertion is correct here since this is internal and annotated

Reply:
> I think an assertion is correct here since this is internal and annotated

we have tests to test these.